src/main/java/org/radarbase/appserver/service/transmitter/EmailNotificationTransmitter.java

```diff
@@ -28,20 +28,16 @@ public EmailNotificationTransmitter(
     @Override
     public void send(Notification notification) {
         if (notification.getUser().getEmailAddress() == null || notification.getUser().getEmailAddress().isBlank()) {
-            log.error("Could not transmit a notification via email because subject {} has no email address.",
+            log.warn("Could not transmit a notification via email because subject {} has no email address.",
                 notification.getUser().getSubjectId());
             return;
         }
-        if (notification.isEmailEnabled()) {
```

Comment:
> Honestly, I do not understand what you did with the protocol. I cannot find the place where the NotificationProtocol is informed whether to use or not use email.
 
Apologies, my bad. I should have included a rationale with the code. The information about whether to use email will come from the project [protocol](https://github.com/RADAR-base/RADAR-aRMT-protocols). This will be in the notifications key (e.g. [here](https://github.com/RADAR-base/RADAR-aRMT-protocols/blob/66a29fac2ff325606bf10494f15361defcf28485/RADAR-EDIFY-KCL-s1/protocol.json#L189)) in each assessment's protocol block.

This is to make the email configuration more granular than enabling it to be deployment-wide. This is because of  these reasons - 

1. Each project may not want this functionality. With protocols, this can be enabled/disabled per project.
2. Within a project, some assessments might be prioritised; a requirement could be that email should only be sent for these priority assessments; others can have disabled email and just have push notifications.
3. Multiple assessments could be scheduled simultaneously in the protocol with the same notification message and title. However, only one notification is usually required for these assessments (to avoid bombarding the user with superfluous notifications). This is achieved by the protocol's Notification block, where we can disable notifications for similar tasks and leave only one enabled (this was recently implemented in #458). A similar arrangement should be made for email notifications. 

We are moving task and protocol scheduling from the aRMT app to the AppServer, so if you are still using scheduling in the app, you might need a minor update there. This would involve reading the new email block in the protocols, and passing this info (`emailEnabled`) when adding notifications to the Appserver (the endpoint should support this as I have also updated the DTO).
Alternatively, you can enable the scheduling through the Appserver using a remote config property, which @mpgxvii can confirm. The Appsever automatically reads the protocol files and schedules tasks and notifications for all the users in the database.

> Note the isEmailEnabled check as first thing in the block?

Yes, you are correct. That is a better approach. If the email is not enabled for this notification (signified by the assessment in the protocol), then we don't execute the email-sending code. You can log a debug message if it was not executed due to emailEnabled=false, which can help debug issues with email sending.



I hope this was clear, but if not, please feel free to Slack me, and we can discuss this further.

Reply:
I reintroduced the `notification.isEmailEnabled()` conditional. We decided to keep both the notification-level and application-level configuration option to enable notifications via email.